lib/parsers/gomod-graph-parser.ts

```diff
@@ -0,0 +1,32 @@
+import {DepGraphBuilder, DepGraph, PkgInfo} from '@snyk/dep-graph';
+
+function parseGoModGraphLine(line: string) {
+  return line
+      .trim()
+      .split(/\s/)
+      .map((item) => item.split('@'));
+}
+
+export function parseGoModGraph(goModGraphOutput: string,
+                                projectName: string,
+                                projectVersion: string = '0.0.0'): DepGraph {
+
+  const rootPkgInfo = {name: projectName, version: projectVersion};
+  const depGraph = new DepGraphBuilder({name: 'gomodules'}, rootPkgInfo);
+
+  for (const line of goModGraphOutput.trim().split('\n')) {
+    const [[parentName, parentVersion], [childName, childVersion]] = parseGoModGraphLine(line);
+
+    const parentPkg: PkgInfo = {name: parentName, version: parentVersion || '0.0.0'};
+    const childPkg: PkgInfo = {name: childName, version: childVersion || '0.0.0'};
+    const parentNodeId = parentName === rootPkgInfo.name ? 'root-node' : `${parentName}@${parentVersion}`;
```

Comment:
if parentVersion is falsy you'll get the wrong value
use same logic as the next line

Reply:
I am not sure I'm following this?
parentVersion is never falsy... (Also added as a default value in the new push)

Reply:
so did you add "|| '0,0,0" in line 20?

Reply:
See this
https://github.com/snyk/snyk-go-parser/pull/21/files#diff-5d5547a431a21d9651a95835ab87f0c4421fe495f9bf1950c7fee1749f94a76aR22